AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/ProcessMutationErrorFromCloudOperation.swift

```diff
@@ -33,68 +43,306 @@ class ProcessMutationErrorFromCloudOperation: Operation {
         log.verbose(#function)
 
         guard !isCancelled else {
-            queryOperation?.cancel()
-            let apiError = DataStoreError.unknown("Operation cancelled", "")
-            finish(result: .failure(apiError))
+            let error = DataStoreError.unknown("Operation cancelled", "")
+            finish(result: .failure(error))
             return
         }
 
-        processConditionalRequestFailed()
-    }
+        guard case let .error(graphQLErrors) = error else {
+            finish(result: .success(nil))
+            return
+        }
 
-    private func processConditionalRequestFailed() {
-        if case let .error(graphQLErrors) = error {
-            // TODO: Check for 'ConflictUnhandled', execute conflict handler
+        guard graphQLErrors.count == 1 else {
+            finish(result: .success(nil))
+            return
+        }
 
-            let hasConditionalRequestFailed = graphQLErrors.contains { (error) -> Bool in
-                if let extensions = error.extensions, case let .string(errorTypeValue) = extensions["errorType"] {
-                    return AppSyncErrorType(errorTypeValue) == .conditionalCheck
-                }
-                return false
-            }
+        guard let graphQLError = graphQLErrors.first else {
+            finish(result: .success(nil))
+            return
+        }
 
-            if hasConditionalRequestFailed {
+        if let extensions = graphQLError.extensions, case let .string(errorTypeValue) = extensions["errorType"] {
+            let errorType = AppSyncErrorType(errorTypeValue)
+            switch errorType {
+            case .conditionalCheck:
                 let payload = HubPayload(eventName: HubPayload.EventName.DataStore.conditionalSaveFailed,
                                          data: mutationEvent)
                 Amplify.Hub.dispatch(to: .dataStore, payload: payload)
+                finish(result: .success(nil))
+            case .conflictUnhandled:
+                processConflictUnhandled(extensions)
+            case .unknown(let errorType):
+                log.debug("Unhandled error with errorType \(errorType)")
+                finish(result: .success(nil))
             }
         }
```

Comment:
Pretty sure we do not have any unit tests that hit this case nor the case where we don't fall into the `if let` statement.. we should add a todo to write some unit tests

Reply:
thanks, operation needs to finish if it doesn't fall into `if let`, updating logic and adding unit tests for both cases